[FreeType] Move the font set cache to FontCache

carlosgcampos opened the pull request:
#### 115d9297ff1ae8ce0d01358d42d3f7bc8b33e577
<pre>
[FreeType] Move the font set cache to FontCache
<a href="https://bugs.webkit.org/show_bug.cgi?id=244195">https://bugs.webkit.org/show_bug.cgi?id=244195</a>

Reviewed by Michael Catanzaro.

With OffscreenCanvas the font set cache can be used from worker threads,
so move it to FontCache to ensure there&apos;s one cache per thread instead.

* Source/WebCore/platform/FreeType.cmake:
* Source/WebCore/platform/graphics/FontCache.h:
* Source/WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:
(WebCore::FontCache::configurePatternForFontDescription):
(WebCore::FontCache::systemFallbackForCharacters):
(WebCore::FontCache::platformPurgeInactiveFontData):
(WebCore::configurePatternForFontDescription): Deleted.
(): Deleted.
(WebCore::CachedFontSet::CachedFontSet): Deleted.
(WebCore::CachedFontSet::bestForCharacters): Deleted.
(WebCore::FallbackFontDescriptionKey::FallbackFontDescriptionKey): Deleted.
(WebCore::FallbackFontDescriptionKey::operator== const): Deleted.
(WebCore::FallbackFontDescriptionKey::operator!= const): Deleted.
(WebCore::FallbackFontDescriptionKey::isHashTableDeletedValue const): Deleted.
(WebCore::add): Deleted.
(WebCore::FallbackFontDescriptionKeyHash::hash): Deleted.
(WebCore::FallbackFontDescriptionKeyHash::equal): Deleted.
(WebCore::systemFallbackFontSetCache): Deleted.
* Source/WebCore/platform/graphics/freetype/FontSetCache.cpp: Added.
(WebCore::FontSetCache::FontSet::FontSet):
(WebCore::FontSetCache::bestForCharacters):
(WebCore::FontSetCache::clear):
* Source/WebCore/platform/graphics/freetype/FontSetCache.h: Added.
(WebCore::FontSetCacheKey::FontSetCacheKey):
(WebCore::FontSetCacheKey::operator== const):
(WebCore::FontSetCacheKey::operator!= const):
(WebCore::FontSetCacheKey::isHashTableDeletedValue const):
(WebCore::add):
(WebCore::FontSetCacheKeyHash::hash):
(WebCore::FontSetCacheKeyHash::equal):

Canonical link: <a href="https://commits.webkit.org/253672@main">https://commits.webkit.org/253672@main</a>
</pre>

<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/63ef8014b9e63ec0719ba0edbea67948274bbba2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/86662 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/30718 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/17598 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/95505 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/149239 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/90647 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/29097 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/25525 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/78864 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/90748 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/92278 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/23509 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/73586 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/23587 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/78514 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/78864 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/66594 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/26878 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/12716 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/26797 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/13731 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/2606 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/28475 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/36591 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/28419 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/33010 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
This probably needs to be backported to 2.38, right?

carlosgcampos committed:
[FreeType] Move the font set cache to FontCache
https://bugs.webkit.org/show_bug.cgi?id=244195

Reviewed by Michael Catanzaro.

With OffscreenCanvas the font set cache can be used from worker threads,
so move it to FontCache to ensure there's one cache per thread instead.

* Source/WebCore/platform/FreeType.cmake:
* Source/WebCore/platform/graphics/FontCache.h:
* Source/WebCore/platform/graphics/freetype/FontCacheFreeType.cpp:
(WebCore::FontCache::configurePatternForFontDescription):
(WebCore::FontCache::systemFallbackForCharacters):
(WebCore::FontCache::platformPurgeInactiveFontData):
(WebCore::configurePatternForFontDescription): Deleted.
(): Deleted.
(WebCore::CachedFontSet::CachedFontSet): Deleted.
(WebCore::CachedFontSet::bestForCharacters): Deleted.
(WebCore::FallbackFontDescriptionKey::FallbackFontDescriptionKey): Deleted.
(WebCore::FallbackFontDescriptionKey::operator== const): Deleted.
(WebCore::FallbackFontDescriptionKey::operator!= const): Deleted.
(WebCore::FallbackFontDescriptionKey::isHashTableDeletedValue const): Deleted.
(WebCore::add): Deleted.
(WebCore::FallbackFontDescriptionKeyHash::hash): Deleted.
(WebCore::FallbackFontDescriptionKeyHash::equal): Deleted.
(WebCore::systemFallbackFontSetCache): Deleted.
* Source/WebCore/platform/graphics/freetype/FontSetCache.cpp: Added.
(WebCore::FontSetCache::FontSet::FontSet):
(WebCore::FontSetCache::bestForCharacters):
(WebCore::FontSetCache::clear):
* Source/WebCore/platform/graphics/freetype/FontSetCache.h: Added.
(WebCore::FontSetCacheKey::FontSetCacheKey):
(WebCore::FontSetCacheKey::operator== const):
(WebCore::FontSetCacheKey::operator!= const):
(WebCore::FontSetCacheKey::isHashTableDeletedValue const):
(WebCore::add):
(WebCore::FontSetCacheKeyHash::hash):
(WebCore::FontSetCacheKeyHash::equal):

Canonical link: https://commits.webkit.org/253672@main
